content/post/the-lab-book.md

```diff
@@ -0,0 +1,87 @@
+
+## What is the lab book?
+
+The lab notebook has been around, mostly unchanged, for hundreds of years. The lab notebook is, most simply, a record of what happens in the lab as a researcher is running experiments. It captures the researcher’s thoughts and ideas, their hypotheses, and their learnings.
+
+If you recall the famous (and perhaps apocryphal) Thomas Edison quote “I have not failed. I've just found 10,000 ways that won't work” you surely must imagine that he kept track of those 10,000 attempts somewhere. The place for tracking those attempts, learning what won’t work, and (hopefully) coming up with ideas for what will, takes place in the lab book.
+
+At the most general level, a lab book should have the following qualities:
+
+* It should record the context of an experiment: the hypothesis being tested and the results observed
+* It should be readable by others: other members of the team should be able to read the entries and be able to replicate the experiment and obtain the same results
+
+Any conclusions that are drawn from a given experiment should reference supporting evidence in the lab book.
+
+## The lab book for data scientists
+
+So, given that all of our data are permanently stored in databases that we can access at any time, why do we need a lab book?
+
+Often, the iteration that a data scientist goes through are opaque to their collaborators -- they may share a working (and reproducible) python notebook, but it’s often impossible to recover the different iterations the DS went through to get to that particular configuration, or what was learned along the way. While using Git can help, the way data scientists use git often does not capture the scientific process of hypothesis-experiment-analysis-conclusion. Instead, you end up with one commit that says “model works” and you don’t have a record of all of the different things the data scientist observed and learned along the way.
+
+When it comes to understanding *_why_* a given model is structured the way it is, you need a history of what was already tried and what worked and what didn’t. If someone on the team wants to pick up where the first practitioner left off, they can avoid lots of wasted effort if they know what was attempted before.
+
+## When to use the lab book?
+
+Have you ever found yourself in a situation where you are working on a gnarly piece of code and you feel like you are just slamming your head into a brick wall over and over and over again? This happens to me pretty frequently and, if you’re around me while I’m going through this, you’ll hear me repeatedly say, aloud, “how on earth could *_that_* be happening?”. At about the third time I say that phrase aloud, I know I need to bust out the lab book.
+
+Software complexity (especially in the data science space) can make it difficult or impossible to reason about what’s happening. For example:
+
+* Working third-party opaque libraries
```

Comment:
A few examples or a very brief explanation (e.g., what is a stochastic process) might be helpful